SwiftFlow/CoreTypes/Store.swift

```diff
@@ -77,44 +89,126 @@ public protocol Store {
      ```
 
      This action creator can then be dispatched as following:
-     ```swift
+
+     ```
      store.dispatch( noteActionCreatore.deleteNote(3) )
      ```
 
      - returns: By default returns the dispatched action, but middlewares can change the
      return type, e.g. to return promises
      */
-    func dispatch(actionCreatorProvider: ActionCreator) -> Any
+    func dispatch(actionCreator: ActionCreator) -> Any
 
     /**
      Dispatches an async action creator to the store. An async action creator generates an
-     action creator asynchronously. Use this method if you want to wait for the state change
-     triggered by the asynchronously generated action creator.
+     action creator asynchronously.
      */
-    func dispatch(asyncActionCreatorProvider: AsyncActionCreator)
+    func dispatch(asyncActionCreator: AsyncActionCreator)
 
     /**
      Dispatches an action and calls the callback as soon as the action has been processed.
      You will receive the updated store state as part of this callback.
 
      Example of dispatching an action and implementing a callback:
-     ```swift
+     
+     ```
      store.dispatch( CounterAction.IncreaseCounter ) { state in
         print("New state: \(state)")
      }
      ```
+     
```

Comment:
Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)